tests/integration/varlink_service_test.py

```diff
@@ -0,0 +1,110 @@
+#
+# Copyright (c) 2020 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+import pytest
+import varlink
+import threading
+
+from libnmstate import netinfo
+from libnmstate.varlink import get_varlink_server
+from libnmstate.schema import Interface
+from libnmstate.schema import InterfaceType
+from libnmstate.schema import InterfaceState
+from libnmstate.schema import InterfaceIPv4
+from libnmstate.schema import InterfaceIPv6
+
+
+VARLINK_INTERFACE = "io.nmstate"
+
+
+@pytest.fixture(
+    params=["unix:@nmstate", "tcp:127.0.0.1:8000"],
+    ids=["unix_socket_connection", "tcp_connection"],
+)
+def server(request):
+    server = get_varlink_server(request.param)
+    thread_server = threading.Thread(target=server.serve_forever, daemon=True)
+    thread_server.start()
+    yield server
+    server.shutdown()
+
+
+def _format_address(server_address):
+    if isinstance(server_address, tuple):
+        return "tcp:" + str(server_address[0]) + ":" + str(server_address[-1])
+    else:
+        return "unix:" + str(server_address)
```

Comment:
```suggestion
        return f"unix:f{server_address}"
```
You can also use format strings, please also check other occurrences like two lines above.